sorc/ncep_post.fd/FDLVL.f

```diff
@@ -795,106 +743,76 @@ SUBROUTINE FDLVL_UV(ITYPE,NFD,HTFD,UFD,VFD)
-!     PROCESS UNTIL ALL NFD FD LEVELS HAVE BEEN PROCESSED.
-!     MOVE ON TO THE NEXT MASS POINT.  
-!     
-!     AVERAGING THE FOUR ABOVE GROUND WINDS TO THE MASS POINT
-!     WAS FOUND TO SMOOTH THE FIELD AND REDUCE THE OCCURRENCE
-!     OF POINT PEAK WINDS FAR IN EXCESS OF THE WINDS AT 
-!     ADJACENT POINTS.  MASS POINT VALUES ARE RETURNED.
-!   .     
-!     
-! PROGRAM HISTORY LOG:
-!   92-12-22  RUSS TREADON
-!   93-11-23  RUSS TREADON - CORRECTED ROUTINE TO COMPUTE
-!             FD LEVELS WITH REPECT TO MEAN SEA LEVEL.
-!   94-01-04  MICHAEL BALDWIN - INCLUDE OPTIONS FOR COMPUTING
-!                               EITHER AGL OR MSL
-!   98-06-15  T BLACK - CONVERSION FROM 1-D TO 2-D
-!   00-01-04  JIM TUCCILLO - MPI VERSION            
-!   02-01-15  MIKE BALDWIN - WRF VERSION
-!   11-12-14  SARAH LU - ADD GOCART AEROSOL AERFD
-!   17-06-01  Y Mao - ADD FD levels for GTG(EDPARM CATEDR MWTURB) and allow 
-!                     levels input from control file
-!   19-09-25  Y MAO - SEPERATE MASS FROM UV
-!                     ALLOW ARRAY OF MASS INPUT TO INTERPOLATE MULTIPLE FIELDS 
-!                     WITH THE SAME LEVELS AT ONE TIME
-!                     DUST=>AERFD CAN BE PROCESSED WHEN NIN=NBIN_DU
-!   20-11-10  JESSE MENG - USE UPP_PHYSICS MODULE
-!     
-! USAGE:    CALL FDLVL_MASS(ITYPE,NFD,PTFD,HTFD,NIN,QIN,QTYPE,QFD)
-!   INPUT ARGUMENT LIST:
-!     ITYPE    - FLAG THAT DETERMINES WHETHER MSL (1) OR AGL (2)
-!                   LEVELS ARE USED.
-!     NFD      - NUMBER OF FD LEVELS
-!     PTFD     - FD PRESSURE LEVELS
-!     HTFD     - FD HEIGHT LEVELS
-!     NIN      - NUMBER OF INPUT FIELDS
-!     QIN      - ARRAY OF MASS POINT VALUE ON MODEL LEVELS
-!     QTYPE    - CHARACTER ARRAY OF VARIABLE TYPE TO DIFFERENTIATE UNDERGROUND INTERPOLATION 
+!> Computes FD level for mass variables.
+!>
+!> This routine computes mass variables (temperature, spec. hum...)
+!> on NFD FD levels. The height of these levels (in meters) is 
+!> passed as an input parameter. The alogrithm proceeds as
+!> follows. (AGL in parentheses)
+!> 
+!> At each mass point move up vertically from the LM-TH (lowest
+!> atmospheric) ETA layer. Find the ETA layers whose  
+!> height (above ground) bounds the target FD level height.
+!> Vertically interpolate to get temperature at this FD
+!> level. Average the four surrounding winds 
+!> to get a mass point wind. Vertically interpolate these 
+!> mass point winds to the target FD level. Continue this 
+!> process until all NFD FD levels have been processed.
+!> Move on to the next mass point.
+!>     
+!> Averaging the four above ground winds to the mass point
+!> was found to smooth the field and reduce the occurrence
+!> of point peak winds far in excess of the winds at 
+!> adjacent points. Mass point values are returned.
+!>
+!> NOTES for Q fields by Y Mao:
+!> The following safety check should be executed by the caller of FDLVL subroutines.
+!> Safety check to avoid tiny QFD values.
+!><pre>
```

Comment:
Instead of using <pre></pre> for code, you should use @code and @endcode . These will present the code in a more attractive format than pre.

Reply:
Thanks again for reviewing. Yes the code now looks awesome!